tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -216,27 +261,52 @@ final class ScalaJSClassEmitter(semantics: Semantics, outputMode: OutputMode,
   private def genES6Constructor(tree: LinkedClass): js.Tree = {
     implicit val pos = tree.pos
 
-    val fieldDefs = genFieldDefs(tree)
-    if (fieldDefs.isEmpty && outputMode == OutputMode.ECMAScript6) {
-      js.Skip()
+    if (tree.kind == ClassKind.JSClass) {
+      val js.Function(params, body) = genConstructorFunForJSClass(tree)
+      js.MethodDef(static = false, js.Ident("constructor"), params, body)
     } else {
-      val superCtorCall = tree.superClass.fold[js.Tree] {
-        js.Skip()(tree.pos)
-      } { parentIdent =>
-        js.Apply(js.Super(), Nil)
-      }
-      val initClassData = outputMode match {
-        case OutputMode.ECMAScript6StrongMode =>
-          js.Assign(js.DotSelect(js.This(), js.Ident("$classData")),
-              envField("d", tree.name.name))
-        case _ =>
-          js.Skip()
+      val fieldDefs = genFieldDefs(tree)
+      if (fieldDefs.isEmpty && outputMode == OutputMode.ECMAScript6) {
+        js.Skip()
+      } else {
+        val superCtorCall = tree.superClass.fold[js.Tree] {
+          js.Skip()(tree.pos)
+        } { parentIdent =>
+          js.Apply(js.Super(), Nil)
+        }
+        val initClassData = outputMode match {
+          case OutputMode.ECMAScript6StrongMode =>
+            js.Assign(js.DotSelect(js.This(), js.Ident("$classData")),
+                envField("d", tree.name.name))
+          case _ =>
+            js.Skip()
+        }
+        js.MethodDef(static = false, js.Ident("constructor"), Nil,
+            js.Block(superCtorCall :: initClassData :: fieldDefs))
       }
-      js.MethodDef(static = false, js.Ident("constructor"), Nil,
-          js.Block(superCtorCall :: initClassData :: fieldDefs))
     }
   }
 
+  private def genConstructorFunForJSClass(tree: LinkedClass): js.Function = {
+    implicit val pos = tree.pos
+
+    require(tree.kind == ClassKind.JSClass)
+
+    val ctor = tree.exportedMembers.map(_.tree) collectFirst {
+      case m: MethodDef
+          if m.static == false && m.name == StringLiteral("constructor") =>
+        m
+    } getOrElse {
+      throw new IllegalArgumentException(
+          s"${tree.encodedName} does not have an exported constructor")
+    }
```

Comment:
Oh, I see. This is not actually exported, the JS ctor just gets classified in the `exportedMembers` field of the `LinkedClass`. How can we (on an IR level) distinguish an exported JS class from a non-exported one (i.e. what is the plan to do so)?

Shouldn't we stick with the `init___` convention and consider the fact that in ES6 the constructor is called "constructor" an implementation detail? It seems odd to have multiple magic names if there is no need to do so...